@kiva/kv-components/vue/KvEditButton.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+	<div>
+		<!-- Edit Button -->
+		<KvButton
+			variant="ghost"
+			class="tw-rounded tw-border tw-border-black"
+			aria-label="Edit"
+			@click="showLightbox"
+		>
+			<KvMaterialIcon :icon="mdiPencil" />
+		</KvButton>
+
+		<!-- Lightbox -->
+		<KvLightbox
+			:visible="visible"
+			:title="title"
+			:prevent-close="preventClose"
+			:variant="variant"
+			@lightbox-closed="hideLightbox"
+		>
+			<slot></slot>
+		</KvLightbox>
+	</div>
+</template>
+
+<script>
+import { mdiPencil } from '@mdi/js';
+import KvButton from './KvButton.vue';
+import KvMaterialIcon from './KvMaterialIcon.vue';
+import KvLightbox from './KvLightbox.vue';
+
+export default {
+	components: {
+		KvButton,
+		KvMaterialIcon,
+		KvLightbox,
+	},
+	props: {
+		/**
+		 * Appearance and role of the lightbox
+		 * @values lightbox, alert
+		 * */
+		variant: {
+			type: String,
+			default: 'lightbox',
+			validator(value) {
+				return ['lightbox', 'alert'].includes(value);
+			},
+		},
+		/**
+		 * The title of the dialog which describes the dialog to screenreaders, and if no
+		 * content is in the `header` slot, will be displayed at the top of the lightbox.
+		 * */
+		title: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * The dialog has no close X button, clicking the screen does not close,
+		 * pressing ESC does not close.
+		 * */
+		preventClose: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	emits: [
+		'lightbox-opened',
+		'lightbox-closed',
+	],
+	setup({ emit }) {
+		const showLightbox = () => {
+			this.visible = true;
```

Comment:
`visible` needs to be a ref, here `this` is undefined.

Reply:
I'm just going to make this component manage state instead of the parent component.
I think it makes sense for the lightbox component itself to have visibility set by a parent component, but for the edit button, the button controls the visibility of the lightbox and that makes integrating the edit button simpler.

If we happen to run into a need to have the lightbox be open by default we can always add support for that afterwards..